packages/cli/src/commands/env/index.ts

```diff
@@ -126,6 +128,24 @@ export default async function main(client: Client) {
   const subArgs = argv._.slice(1);
   const { subcommand, args } = getSubcommand(subArgs, COMMAND_CONFIG);
   const { output, config } = client;
+
+  const environmentArg = argv['--environment']?.toLowerCase();
+  let target = ProjectEnvTarget.Development;
+  if (environmentArg && environmentArg !== 'development') {
+    if (environmentArg === 'preview') {
+      target = ProjectEnvTarget.Preview;
+    } else if (environmentArg === 'production') {
+      target = ProjectEnvTarget.Production;
+    } else {
+      output.error(
+        `Invalid environment \`${chalk.cyan(
+          environmentArg
+        )}\`. Options: development | preview | production`
+      );
+      return 1;
+    }
+  }
```

Comment:
Converting strings to enums is messy business. There's a helper that makes this a bit easier and will work with any future changes to `ProjectEnvTarget`. I think you could do something like this:

```
// TODO: you'll have to export the `envTargets` function in this file
import { isValidEnvTarget, envTargets } from '../../util/env/env-target.ts';

// ...

function parseEnvironmentTarget(environment): ProjectEnvTarget | undefined {
  if (isValidEnvTarget(environment)) {
    return environment;
  }
  return undefined;
}

// ...

const target = environmentArg
  ? parseEnvironmentTarget(environmentArg)
  : ProjectEnvTarget.Development;

if (!target) {
  output.error(
    `Invalid environment \`${chalk.cyan(
      environmentArg
    )}\`. Valid options: ${envTargets()}`
  );
  return 1;
}
```